language/move-lang/src/lib.rs

```diff
@@ -724,18 +724,27 @@ fn ensure_targets_deps_dont_intersect(
     deps: &mut Vec<&'static str>,
     sources_shadow_deps: bool,
 ) -> anyhow::Result<()> {
-    let target_set = targets.iter().collect::<BTreeSet<_>>();
-    let dep_set = deps.iter().collect::<BTreeSet<_>>();
-    let intersection = target_set
-        .intersection(&dep_set)
-        .cloned()
-        .cloned()
-        .collect::<Vec<_>>();
+    /// Canonicalize a file path.
+    fn canonicalize(path: &str) -> String {
+        match std::fs::canonicalize(path) {
+            Ok(s) => s.to_string_lossy().to_string(),
+            Err(_) => path.to_owned(),
+        }
+    }
```

Comment:
Thanks a lot for the fix! Double checking with @tnowacki as the canonicalization is not done in the original intersection check, so just want to make sure that the "no canonicalization" is not by design.